File: js/systems/ChallengeSystem.js
// Challenge System
class ChallengeSystem {
    constructor(gameManager) {
        this.gameManager = gameManager;
        this.challenges = challengeDatabase;
        this.currentChallenge = null;
        
        // Challenge gameplay state
        this.playerProgress = 0;
        this.aiProgress = {};
        this.targetProgress = 100;
        this.tapIncrement = 5;
        this.aiIncrementBase = 3;
        this.challengeActive = false;
        this.competingTribes = [];
        this.competingIndividuals = [];
        this.individualWinner = null;
        this.tribeWinner = null;
        this.immunePlayers = [];
    }
    
    /**
     * Initialize the challenge system
     */
    initialize() {
        this.currentChallenge = null;
        this.playerProgress = 0;
        this.aiProgress = {};
        this.challengeActive = false;
        this.competingTribes = [];
        this.competingIndividuals = [];
        this.individualWinner = null;
        this.tribeWinner = null;
        this.immunePlayers = [];
    }
    
    /**
     * Start a new challenge
     */
    startChallenge() {
        // Determine challenge type based on game phase
        const challengeType = this.gameManager.getGamePhase() === "preMerge" ? "tribe" : "individual";
        
        // Filter challenges by type
        const eligibleChallenges = this.challenges.filter(c => c.type === challengeType);
        
        // Select a random challenge
        this.currentChallenge = eligibleChallenges[getRandomInt(0, eligibleChallenges.length - 1)];
        
        // Reset progress values
        this.playerProgress = 0;
        this.aiProgress = {};
        this.challengeActive = true;
        
        // Determine competitors
        if (challengeType === "tribe") {
            this.competingTribes = this.gameManager.getTribes();
            this.competingIndividuals = [];
            
            // Initialize tribe progress
            this.competingTribes.forEach(tribe => {
                this.aiProgress[tribe.tribeName] = 0;
            });
        } else {
            this.competingTribes = [];
            this.competingIndividuals = [];
            
            // Add all remaining survivors
            this.gameManager.getTribes().forEach(tribe => {
                tribe.members.forEach(member => {
                    this.competingIndividuals.push(member);
                    if (!member.isPlayer) {
                        this.aiProgress[member.name] = 0;
                    }
                });
            });
        }
        
        // Set up UI
        this.updateChallengeUI();
        
        // Start AI progress updates
        this.aiUpdateInterval = setInterval(() => this.updateAIProgress(), 500);
    }
    
    /**
     * Update the challenge UI
     */
    updateChallengeUI() {
        const challengeTitle = document.getElementById('challenge-title');
        const challengeDescription = document.getElementById('challenge-description');
        const challengeProgress = document.getElementById('challenge-progress-bar');
        const challengeButton = document.getElementById('challenge-button');
        
        if (challengeTitle && this.currentChallenge) {
            challengeTitle.textContent = this.currentChallenge.title;
        }
        
        if (challengeDescription && this.currentChallenge) {
            challengeDescription.textContent = this.currentChallenge.description;
        }
        
        if (challengeProgress) {
            challengeProgress.style.width = formatProgressWidth(this.playerProgress, this.targetProgress);
        }
        
        if (challengeButton) {
            if (!this.challengeActive) {
                challengeButton.textContent = "Start Challenge";
                challengeButton.disabled = false;
            } else {
                challengeButton.textContent = "Tap!";
                challengeButton.disabled = false;
            }
        }
        
        // Update competitors display
        this.updateCompetitorsDisplay();
    }
    
    /**
     * Update competitors display in the UI
     */
    updateCompetitorsDisplay() {
        const competitorsContainer = document.getElementById('competitors-container');
        if (!competitorsContainer) return;
        
        // Clear container
        clearChildren(competitorsContainer);
        
        if (this.currentChallenge.type === "tribe") {
            // Tribe challenge display
            this.competingTribes.forEach(tribe => {
                const tribeBox = createElement('div', { className: 'competitor-box' });
                
                const tribeName = createElement('div', { 
                    className: 'competitor-name',
                    textContent: tribe.tribeName
                });
                tribeName.style.color = tribe.tribeColor;
                
                const progressBar = createElement('div', { className: 'progress-bar' });
                const progressFill = createElement('div', { className: 'progress-fill' });
                progressFill.style.width = formatProgressWidth(this.aiProgress[tribe.tribeName] || 0, this.targetProgress);
                progressBar.appendChild(progressFill);
                
                tribeBox.appendChild(tribeName);
                tribeBox.appendChild(progressBar);
                
                competitorsContainer.appendChild(tribeBox);
            });
        } else {
            // Individual challenge display
            this.competingIndividuals.forEach(individual => {
                const individualBox = createElement('div', { className: 'competitor-box' });
                
                const individualName = createElement('div', { 
                    className: 'competitor-name',
                    textContent: individual.name
                });
                
                if (individual.isPlayer) {
                    individualName.style.color = '#3182ce'; // Blue for player
                }
                
                const progressBar = createElement('div', { className: 'progress-bar' });
                const progressFill = createElement('div', { className: 'progress-fill' });
                progressFill.style.width = formatProgressWidth(
                    individual.isPlayer ? this.playerProgress : (this.aiProgress[individual.name] || 0),
                    this.targetProgress
                );
                progressBar.appendChild(progressFill);
                
                individualBox.appendChild(individualName);
                individualBox.appendChild(progressBar);
                
                competitorsContainer.appendChild(individualBox);
            });
        }
    }
    
    /**
     * Handle challenge button press
     */
    onChallengeButtonPressed() {
        if (!this.challengeActive) {
            this.startChallenge();
            return;
        }
        
        // Increment player progress
        this.playerProgress += this.tapIncrement;
        
        // Update progress bar
        const challengeProgress = document.getElementById('challenge-progress-bar');
        if (challengeProgress) {
            challengeProgress.style.width = formatProgressWidth(this.playerProgress, this.targetProgress);
        }
        
        // Update player progress in competitors display
        this.updateCompetitorsDisplay();
        
        // Check if challenge is complete
        if (this.playerProgress >= this.targetProgress) {
            this.completeChallenge(true);
        }
    }
    
    /**
     * Update AI competitors during challenge
     */
    updateAIProgress() {
        if (!this.challengeActive) return;
        
        // Calculate AI progress increment based on challenge type
        if (this.currentChallenge.type === "tribe") {
            this.updateTribeChallenge();
        } else {
            this.updateIndividualChallenge();
        }
        
        // Update the UI
        this.updateCompetitorsDisplay();
    }
    
    /**
     * Update tribe challenge progress
     */
    updateTribeChallenge() {
        // Find player's tribe
        const playerTribe = this.gameManager.getPlayerTribe();
        
        // Update progress for other tribes
        this.competingTribes.forEach(tribe => {
            if (tribe === playerTribe) return;
            
            // Calculate progress based on tribe stats
            let statMultiplier = 1.0;
            switch (this.currentChallenge.primaryStat) {
                case "physical":
                    statMultiplier = this.getTribeAverageStat(tribe, "physical") / 100;
                    break;
                case "mental":
                    statMultiplier = this.getTribeAverageStat(tribe, "mental") / 100;
                    break;
                case "personality":
                    statMultiplier = this.getTribeAverageStat(tribe, "personality") / 100;
                    break;
            }
            
            // Apply resource factor
            const resourceFactor = this.getTribeResourceFactor(tribe);
            
            // Calculate increment
            let increment = this.aiIncrementBase * statMultiplier * resourceFactor * this.currentChallenge.difficulty;
            
            // Add some randomness
            increment *= 0.8 + Math.random() * 0.4; // 0.8 to 1.2
            
            // Update tribe progress
            if (!this.aiProgress[tribe.tribeName]) {
                this.aiProgress[tribe.tribeName] = 0;
            }
            this.aiProgress[tribe.tribeName] += increment;
            
            // Check if any tribe completed the challenge
            if (this.aiProgress[tribe.tribeName] >= this.targetProgress) {
                this.tribeWinner = tribe;
                this.completeChallenge(false);
            }
        });
    }
    
    /**
     * Update individual challenge progress
     */
    updateIndividualChallenge() {
        const playerSurvivor = this.gameManager.getPlayerSurvivor();
        
        // Update progress for other survivors
        this.competingIndividuals.forEach(survivor => {
            if (survivor.isPlayer) return;
            
            // Calculate progress based on survivor stats
            let statMultiplier = 1.0;
            switch (this.currentChallenge.primaryStat) {
                case "physical":
                    statMultiplier = survivor.physicalStat / 100;
                    break;
                case "mental":
                    statMultiplier = survivor.mentalStat / 100;
                    break;
                case "personality":
                    statMultiplier = survivor.personalityStat / 100;
                    break;
            }
            
            // Calculate increment
            let increment = this.aiIncrementBase * statMultiplier * this.currentChallenge.difficulty;
            
            // Add some randomness
            increment *= 0.8 + Math.random() * 0.4; // 0.8 to 1.2
            
            // Update survivor progress
            if (!this.aiProgress[survivor.name]) {
                this.aiProgress[survivor.name] = 0;
            }
            this.aiProgress[survivor.name] += increment;
            
            // Check if any survivor completed the challenge
            if (this.aiProgress[survivor.name] >= this.targetProgress) {
                this.individualWinner = survivor;
                this.completeChallenge(false);
            }
        });
    }
    
    /**
     * Complete the challenge and determine winner
     * @param {boolean} playerWon - Whether the player won
     */
    completeChallenge(playerWon) {
        this.challengeActive = false;
        clearInterval(this.aiUpdateInterval);
        
        // Disable challenge button
        const challengeButton = document.getElementById('challenge-button');
        if (challengeButton) {
            challengeButton.disabled = true;
        }
        
        if (this.currentChallenge.type === "tribe") {
            this.completeTribeChallenge(playerWon);
        } else {
            this.completeIndividualChallenge(playerWon);
        }
        
        // Show continue button
        setTimeout(() => this.showContinueButton(), 2000);
    }
    
    /**
     * Complete tribe challenge
     * @param {boolean} playerWon - Whether the player's tribe won
     */
    completeTribeChallenge(playerWon) {
        const playerTribe = this.gameManager.getPlayerTribe();
        
        if (playerWon) {
            this.tribeWinner = playerTribe;
        }
        
        // Apply immunity to winning tribe
        this.immunePlayers = [];
        this.tribeWinner.members.forEach(member => {
            member.hasImmunity = true;
            this.immunePlayers.push(member);
        });
        
        // Apply resource depletion penalty to losing tribe(s)
        this.gameManager.getTribes().forEach(tribe => {
            if (tribe !== this.tribeWinner) {
                // Tribe lost challenge - deplete resources
                this.depleteTribalResources(tribe);
            }
        });
        
        // Show result
        const challengeDescription = document.getElementById('challenge-description');
        if (challengeDescription) {
            let resultText = `${this.tribeWinner.tribeName} Tribe wins immunity!`;
            
            if (this.tribeWinner === playerTribe) {
                resultText += "\nYour tribe is safe from Tribal Council tonight.";
            } else {
                resultText += "\nYour tribe must attend Tribal Council tonight.";
                resultText += "\nLosing the challenge has depleted some of your tribe's resources.";
            }
            
            challengeDescription.textContent = resultText;
        }
    }
    
    /**
     * Deplete resources for a tribe that lost a challenge
     * @param {Object} tribe - The tribe that lost the challenge
     */
    depleteTribalResources(tribe) {
        // Lose 10-20% of resources randomly
        const resourceLoss = 10 + Math.floor(Math.random() * 11);
        
        // Apply depletion to each resource
        tribe.fire = Math.max(0, tribe.fire - Math.floor(tribe.fire * resourceLoss / 100));
        tribe.water = Math.max(0, tribe.water - Math.floor(tribe.water * resourceLoss / 100));
        tribe.food = Math.max(0, tribe.food - Math.floor(tribe.food * resourceLoss / 100));
    }
    
    /**
     * Complete individual challenge
     * @param {boolean} playerWon - Whether the player won
     */
    completeIndividualChallenge(playerWon) {
        const playerSurvivor = this.gameManager.getPlayerSurvivor();
        
        if (playerWon) {
            this.individualWinner = playerSurvivor;
        }
        
        // Apply immunity to winner
        this.immunePlayers = [this.individualWinner];
        this.individualWinner.hasImmunity = true;
        
        // Show result
        const challengeDescription = document.getElementById('challenge-description');
        if (challengeDescription) {
            let resultText = `${this.individualWinner.name} wins individual immunity!`;
            
            if (this.individualWinner === playerSurvivor) {
                resultText += "\nYou are safe from Tribal Council tonight.";
            }
            
            challengeDescription.textContent = resultText;
        }
    }
    
    /**
     * Show continue button after challenge completion
     */
    showContinueButton() {
        const challengeButton = document.getElementById('challenge-button');
        if (challengeButton) {
            challengeButton.textContent = "Continue";
            challengeButton.disabled = false;
            
            // Create a new event listener for the continue button
            challengeButton.onclick = () => {
                // Determine which state to go to next
                let nextState;
                
                // In pre-merge phase with tribe challenges:
                if (this.currentChallenge.type === "tribe" && 
                    this.gameManager.getGamePhase() === "preMerge") {
                    
                    if (this.tribeWinner === this.gameManager.getPlayerTribe()) {
                        // Player's tribe won immunity - skip tribal council
                        nextState = "camp";
                    } else {
                        // Player's tribe lost immunity - go to tribal council
                        nextState = "tribalCouncil";
                    }
                } 
                // In post-merge phase with individual challenges:
                else if (this.currentChallenge.type === "individual") {
                    // Everyone goes to tribal, but some players have immunity
                    nextState = "tribalCouncil";
                } 
                // Default fallback
                else {
                    nextState = "tribalCouncil";
                }
                
                // Proceed to next state
                this.gameManager.advanceDay();
                this.gameManager.setGameState(nextState);
            };
        }
    }
    
    /**
     * Get average stat for a tribe
     * @param {Object} tribe - The tribe
     * @param {string} statType - The stat type ("physical", "mental", "personality")
     * @returns {number} The average stat value
     */
    getTribeAverageStat(tribe, statType) {
        if (!tribe || !tribe.members || tribe.members.length === 0) return 50;
        
        let total = 0;
        tribe.members.forEach(member => {
            if (statType === "physical") total += member.physicalStat;
            else if (statType === "mental") total += member.mentalStat;
            else if (statType === "personality") total += member.personalityStat;
        });
        
        return total / tribe.members.length;
    }
    
    /**
     * Get resource factor for a tribe
     * @param {Object} tribe - The tribe
     * @returns {number} Resource factor (0.8 to 1.2)
     */
    getTribeResourceFactor(tribe) {
        if (!tribe) return 1.0;
        
        // Calculate based on fire, water, food
        const resources = (tribe.fire + tribe.water + tribe.food) / 3;
        
        // Map resources to factor (50 = 1.0, 0 = 0.8, 100 = 1.2)
        return 0.8 + (resources / 250);
    }
    
    /**
     * Get list of players with immunity
     * @returns {Array} Array of immune players
     */
    getImmunePlayers() {
        return this.immunePlayers;
    }
}

File: Assets/Scripts/Systems/ChallengeSystem.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;
using TMPro;

/// <summary>
/// Manages immunity challenges and their outcomes
/// </summary>
public class ChallengeSystem : MonoBehaviour
{
    [SerializeField] private List<Challenge> _challenges = new List<Challenge>();
    [SerializeField] private Challenge _currentChallenge;
    
    // UI Elements
    [SerializeField] private Slider _progressBar;
    [SerializeField] private TMP_Text _challengeTitle;
    [SerializeField] private TMP_Text _challengeDescription;
    [SerializeField] private TMP_Text _challengeResult;
    [SerializeField] private Button _challengeButton;
    
    // Challenge gameplay state
    private float _playerProgress = 0f;
    private float _aiProgress = 0f;
    private float _targetProgress = 100f;
    private float _tapIncrement = 5f;
    private float _aiIncrementBase = 3f;
    private bool _challengeActive = false;
    private List<TribeData> _competingTribes = new List<TribeData>();
    private List<SurvivorData> _competingIndividuals = new List<SurvivorData>();
    private SurvivorData _individualWinner;
    private TribeData _tribeWinner;
    
    [System.Serializable]
    public class Challenge
    {
        public string Title;
        public string Description;
        public ChallengeType Type;
        public StatType PrimaryStat;
        public float Difficulty; // 0-1 multiplier for AI performance
        
        public Challenge(string title, string description, ChallengeType type, StatType stat, float difficulty)
        {
            Title = title;
            Description = description;
            Type = type;
            PrimaryStat = stat;
            Difficulty = Mathf.Clamp01(difficulty);
        }
    }
    
    public enum ChallengeType
    {
        Tribe,
        Individual
    }
    
    public enum StatType
    {
        Physical,
        Mental,
        Personality
    }
    
    private void Awake()
    {
        InitializeChallenges();
    }
    
    private void InitializeChallenges()
    {
        // Add physical challenges
        _challenges.Add(new Challenge(
            "Obstacle Course Relay", 
            "Teams must race through a challenging obstacle course, passing a baton between tribe members.",
            ChallengeType.Tribe, 
            StatType.Physical, 
            0.7f
        ));
        
        _challenges.Add(new Challenge(
            "Bucket Brigade", 
            "Fill your tribe's container by passing water buckets down a line. Spill as little as possible!",
            ChallengeType.Tribe, 
            StatType.Physical, 
            0.6f
        ));
        
        _challenges.Add(new Challenge(
            "Blindfolded Maze", 
            "Navigate a complex maze while blindfolded, guided only by your tribe's voice commands.",
            ChallengeType.Tribe, 
            StatType.Physical, 
            0.8f
        ));
        
        // Add mental challenges
        _challenges.Add(new Challenge(
            "Memory Match", 
            "Memorize and match symbols on a board. The tribe with the most matches wins.",
            ChallengeType.Tribe, 
            StatType.Mental, 
            0.7f
        ));
        
        _challenges.Add(new Challenge(
            "Puzzle Assembly", 
            "Race to assemble a complex 3D puzzle that forms your tribe's logo.",
            ChallengeType.Tribe, 
            StatType.Mental, 
            0.6f
        ));
        
        _challenges.Add(new Challenge(
            "Word Scramble", 
            "Unscramble a series of words related to survival, then use the highlighted letters to solve a final phrase.",
            ChallengeType.Tribe, 
            StatType.Mental, 
            0.5f
        ));
        
        // Add physical individual challenges
        _challenges.Add(new Challenge(
            "Endurance Hold", 
            "Hold your position on an increasingly unstable platform. Last person standing wins immunity.",
            ChallengeType.Individual, 
            StatType.Physical, 
            0.7f
        ));
        
        _challenges.Add(new Challenge(
            "Balance Beam", 
            "Cross a series of narrow balance beams while collecting keys. First to finish wins immunity.",
            ChallengeType.Individual, 
            StatType.Physical, 
            0.6f
        ));
        
        // Add mental individual challenges
        _challenges.Add(new Challenge(
            "Memory Sequence", 
            "Memorize and repeat an ever-growing sequence of symbols. Last person standing wins immunity.",
            ChallengeType.Individual, 
            StatType.Mental, 
            0.8f
        ));
        
        _challenges.Add(new Challenge(
            "Survivor Sudoku", 
            "Complete a Survivor-themed number puzzle. First to finish correctly wins immunity.",
            ChallengeType.Individual, 
            StatType.Mental, 
            0.7f
        ));
    }
    
    /// <summary>
    /// Start a new challenge
    /// </summary>
    public void StartChallenge()
    {
        // Determine challenge type based on game phase
        ChallengeType challengeType = GameManager.Instance.GetCurrentPhase() == GameManager.GamePhase.PreMerge 
            ? ChallengeType.Tribe 
            : ChallengeType.Individual;
            
        // Filter challenges by type
        List<Challenge> eligibleChallenges = _challenges.FindAll(c => c.Type == challengeType);
        
        // Select a random challenge
        _currentChallenge = eligibleChallenges[Random.Range(0, eligibleChallenges.Count)];
        
        // Reset progress values
        _playerProgress = 0f;
        _aiProgress = 0f;
        _challengeActive = true;
        
        // Determine competitors
        if (challengeType == ChallengeType.Tribe)
        {
            _competingTribes = GameManager.Instance.GetTribes();
            _competingIndividuals.Clear();
        }
        else
        {
            _competingTribes.Clear();
            _competingIndividuals = new List<SurvivorData>();
            
            // Add all remaining survivors
            foreach (var tribe in GameManager.Instance.GetTribes())
            {
                _competingIndividuals.AddRange(tribe.Members);
            }
        }
        
        // Set up UI
        if (_challengeTitle != null)
            _challengeTitle.text = _currentChallenge.Title;
            
        if (_challengeDescription != null)
            _challengeDescription.text = _currentChallenge.Description;
            
        if (_challengeResult != null)
            _challengeResult.gameObject.SetActive(false);
            
        if (_progressBar != null)
        {
            _progressBar.value = 0;
            _progressBar.maxValue = _targetProgress;
        }
        
        if (_challengeButton != null)
        {
            _challengeButton.interactable = true;
            _challengeButton.onClick.RemoveAllListeners();
            _challengeButton.onClick.AddListener(OnChallengeButtonPressed);
        }
    }
    
    private void OnChallengeButtonPressed()
    {
        if (!_challengeActive) return;
        
        // Increment player progress
        _playerProgress += _tapIncrement;
        
        // Update progress bar
        if (_progressBar != null)
        {
            _progressBar.value = _playerProgress;
        }
        
        // Check if challenge is complete
        if (_playerProgress >= _targetProgress)
        {
            CompleteChallenge(true);
        }
    }
    
    /// <summary>
    /// Update AI competitors during challenge
    /// </summary>
    private void Update()
    {
        if (!_challengeActive) return;
        
        // Only update AI every 0.5 seconds
        if (Time.frameCount % 30 != 0) return;
        
        // Calculate AI progress increment based on challenge type
        if (_currentChallenge.Type == ChallengeType.Tribe)
        {
            UpdateTribeChallenge();
        }
        else
        {
            UpdateIndividualChallenge();
        }
    }
    
    private void UpdateTribeChallenge()
    {
        // Find player's tribe
        TribeData playerTribe = GameManager.Instance.GetPlayerTribe();
        
        // Update progress for other tribes
        foreach (var tribe in _competingTribes)
        {
            if (tribe == playerTribe) continue;
            
            // Calculate progress based on tribe stats
            float statMultiplier = 1.0f;
            switch (_currentChallenge.PrimaryStat)
            {
                case StatType.Physical:
                    statMultiplier = tribe.GetAveragePhysicalStat() / 100f;
                    break;
                case StatType.Mental:
                    statMultiplier = tribe.GetAverageMentalStat() / 100f;
                    break;
                case StatType.Personality:
                    statMultiplier = tribe.GetAveragePersonalityStat() / 100f;
                    break;
            }
            
            // Apply resource factor
            float resourceFactor = tribe.GetResourceFactor();
            
            // Calculate increment
            float increment = _aiIncrementBase * statMultiplier * resourceFactor * _currentChallenge.Difficulty;
            
            // Add some randomness
            increment *= Random.Range(0.8f, 1.2f);
            
            // Update tribe progress
            tribe.FireChange = Mathf.RoundToInt(increment);
            
            // Check if any tribe completed the challenge
            if (tribe.FireChange >= _targetProgress)
            {
                _tribeWinner = tribe;
                CompleteChallenge(false);
                break;
            }
        }
        
        // Check if player's tribe is close to winning/losing for dramatic effect
        if (_challengeActive && playerTribe != null)
        {
            float closestTribeProgress = 0f;
            foreach (var tribe in _competingTribes)
            {
                if (tribe != playerTribe && tribe.FireChange > closestTribeProgress)
                {
                    closestTribeProgress = tribe.FireChange;
                }
            }
            
            // If any tribe is within 80% of completion, show warning
            if (closestTribeProgress >= _targetProgress * 0.8f && _playerProgress < closestTribeProgress)
            {
                if (_challengeDescription != null)
                {
                    _challengeDescription.text = "Another tribe is close to finishing! Tap faster!";
                }
            }
        }
    }
    
    private void UpdateIndividualChallenge()
    {
        SurvivorData playerSurvivor = GameManager.Instance.GetPlayerSurvivor();
        
        // Update progress for other survivors
        foreach (var survivor in _competingIndividuals)
        {
            if (survivor == playerSurvivor) continue;
            
            // Calculate progress based on survivor stats
            float statMultiplier = 1.0f;
            switch (_currentChallenge.PrimaryStat)
            {
                case StatType.Physical:
                    statMultiplier = survivor.PhysicalStat / 100f;
                    break;
                case StatType.Mental:
                    statMultiplier = survivor.MentalStat / 100f;
                    break;
                case StatType.Personality:
                    statMultiplier = survivor.PersonalityStat / 100f;
                    break;
            }
            
            // Calculate increment
            float increment = _aiIncrementBase * statMultiplier * _currentChallenge.Difficulty;
            
            // Add some randomness
            increment *= Random.Range(0.8f, 1.2f);
            
            // Track survivor progress
            survivor.PhysicalStat = Mathf.RoundToInt(survivor.PhysicalStat + increment);
            
            // Check if any survivor completed the challenge
            if (survivor.PhysicalStat >= _targetProgress)
            {
                _individualWinner = survivor;
                CompleteChallenge(false);
                break;
            }
        }
    }
    
    private void CompleteChallenge(bool playerWon)
    {
        _challengeActive = false;
        
        if (_challengeButton != null)
        {
            _challengeButton.interactable = false;
        }
        
        if (_currentChallenge.Type == ChallengeType.Tribe)
        {
            CompleteTribeChallenge(playerWon);
        }
        else
        {
            CompleteIndividualChallenge(playerWon);
        }
        
        // Show continue button after 2 seconds
        StartCoroutine(ShowContinueButton());
    }
    
    private void CompleteTribeChallenge(bool playerWon)
    {
        TribeData playerTribe = GameManager.Instance.GetPlayerTribe();
        
        if (playerWon)
        {
            _tribeWinner = playerTribe;
        }
        
        // Apply immunity to winning tribe
        foreach (var member in _tribeWinner.Members)
        {
            member.GiveImmunity();
        }
        
        // Show result
        if (_challengeResult != null)
        {
            _challengeResult.gameObject.SetActive(true);
            _challengeResult.text = $"{_tribeWinner.TribeName} Tribe wins immunity!";
            
            if (_tribeWinner == playerTribe)
            {
                _challengeResult.text += "\nYour tribe is safe from Tribal Council tonight.";
            }
            else
            {
                _challengeResult.text += "\nYour tribe must attend Tribal Council tonight.";
            }
        }
    }
    
    private void CompleteIndividualChallenge(bool playerWon)
    {
        SurvivorData playerSurvivor = GameManager.Instance.GetPlayerSurvivor();
        
        if (playerWon)
        {
            _individualWinner = playerSurvivor;
        }
        
        // Apply immunity to winner
        _individualWinner.GiveImmunity();
        
        // Show result
        if (_challengeResult != null)
        {
            _challengeResult.gameObject.SetActive(true);
            _challengeResult.text = $"{_individualWinner.Name} wins individual immunity!";
            
            if (_individualWinner == playerSurvivor)
            {
                _challengeResult.text += "\nYou are safe from Tribal Council tonight.";
            }
        }
    }
    
    private IEnumerator ShowContinueButton()
    {
        yield return new WaitForSeconds(2f);
        
        // Add continue button
        if (_challengeButton != null)
        {
            _challengeButton.interactable = true;
            _challengeButton.GetComponentInChildren<TMP_Text>().text = "Continue";
            _challengeButton.onClick.RemoveAllListeners();
            _challengeButton.onClick.AddListener(() => {
                // Determine which state to go to next
                GameManager.GameState nextState = GameManager.GameState.TribalCouncil;
                
                // If player's tribe won immunity in pre-merge phase
                if (_currentChallenge.Type == ChallengeType.Tribe && 
                    _tribeWinner == GameManager.Instance.GetPlayerTribe() &&
                    GameManager.Instance.GetCurrentPhase() == GameManager.GamePhase.PreMerge)
                {
                    // Skip tribal council, go to next day
                    GameManager.Instance.GetPlayerTribe().GetPlayerMember().RemoveImmunity();
                    GameManager.Instance.SetGameState(GameManager.GameState.Camp);
                }
                else
                {
                    // Proceed to tribal council
                    GameManager.Instance.SetGameState(nextState);
                }
            });
        }
    }
    
    /// <summary>
    /// Get list of survivors who have immunity
    /// </summary>
    public List<SurvivorData> GetImmunePlayers()
    {
        List<SurvivorData> immunePlayers = new List<SurvivorData>();
        
        foreach (var tribe in GameManager.Instance.GetTribes())
        {
            foreach (var member in tribe.Members)
            {
                if (member.HasImmunity)
                {
                    immunePlayers.Add(member);
                }
            }
        }
        
        return immunePlayers;
    }
    
    /// <summary>
    /// Reset immunity status for all survivors
    /// </summary>
    public void ResetImmunity()
    {
        foreach (var tribe in GameManager.Instance.GetTribes())
        {
            foreach (var member in tribe.Members)
            {
                member.RemoveImmunity();
            }
        }
    }
}
